### PR TITLE
[New] `jsx-tag-spacing`: Add `multiline-always` option

### LINKED ISSUE #3210: Is it possible to catch a single prop where the prop is a multiline object?
Specifically the formatting that I want to catch is this:
```
<div style={{
   background: 'red'
}}
>
```

The JSX rules I have so far as as follows:
```
    "react/jsx-closing-tag-location": [2],
    "react/jsx-curly-spacing": [2, "never"],
    "react/jsx-curly-newline": [2, "never"],
    "react/jsx-closing-bracket-location": [2, "tag-aligned"],
    "react/jsx-equals-spacing": [2, "never"],
    "react/jsx-indent": [2, 2, { "checkAttributes": true, "indentLogicalExpressions": true }],
    "react/jsx-indent-props": [2, 2],
    "react/jsx-max-props-per-line": [1, { "maximum": { "single": 3, "multiline": 1 } }],
    "react/jsx-no-useless-fragment": [2],
    "react/jsx-one-expression-per-line": [2, { "allow": "single-child" }],
    "react/jsx-tag-spacing": [2, { "beforeSelfClosing": "always", "beforeClosing": "never" }],
```

The big issue is this is a single prop, that happens to be an object. I would love to enforce that this prop _must_ be put on a newline but I cannot see a way to achieve that from reading the JSX rules in depth. Any suggestions?

The _correct_ formatting I'd like to enforce is:
```
<div
  style={{
    background: 'red'
  }}
>
```

### PR DESCRIPTION
**Pull request by Nokel81**
- This option enforces that the closingTag or the selfClosingTag is on a
  new line if the entire tag is a multiline tag

Fixes #3210.

### CODE REVIEW

File: tests/lib/rules/jsx-tag-spacing.js [hunk 1]
```diff
@@ -302,6 +345,60 @@ ruleTester.run('jsx-tag-spacing', rule, {
       options: beforeSelfClosingOptions('never'),
       errors: [{ messageId: 'beforeSelfCloseNoSpace' }],
     },
+    {
+      code: `
+        <App
+          foo={bar}/>`,
+      output: `
+        <App
+          foo={bar}\n/>`,
```
Comment (ljharb) [inline tests/lib/rules/jsx-tag-spacing.js:354]:
```suggestion
          foo={bar}
/>`,
```

File: tests/lib/rules/jsx-tag-spacing.js [hunk 2]
```diff
@@ -302,6 +345,60 @@ ruleTester.run('jsx-tag-spacing', rule, {
...
+      options: beforeSelfClosingOptions('multiline-always'),
+      errors: [{ messageId: 'beforeSelfCloseNeedNewline' }],
+    },
+    {
+      code: `
+        <App
+          foo={bar} />`,
+      output: `
+        <App
+          foo={bar} \n/>`,
```
Comment (ljharb) [inline tests/lib/rules/jsx-tag-spacing.js:364]:
```suggestion
          foo={bar}${' '}
/>`,
```

File: tests/lib/rules/jsx-tag-spacing.js [hunk 3]
```diff
@@ -302,6 +345,60 @@ ruleTester.run('jsx-tag-spacing', rule, {
...
+        <App
+          foo={bar}
+          blat >
+          hello
+        </App>
+      `,
+      output: `
+        <App
+          foo={bar}
+          blat \n>
```
Comment (ljharb) [inline tests/lib/rules/jsx-tag-spacing.js:379]:
```suggestion
          blat${' '}
>
```

File: tests/lib/rules/jsx-tag-spacing.js [hunk 4]
```diff
@@ -302,6 +345,60 @@ ruleTester.run('jsx-tag-spacing', rule, {
...
+    {
+      code: `
+        <App
+          foo={bar}>
+          hello
+        </App>
+      `,
+      output: `
+        <App
+          foo={bar}\n>
```
Comment (ljharb) [inline tests/lib/rules/jsx-tag-spacing.js:395]:
```suggestion
          foo={bar}
>
```

**Review by ljharb — APPROVED** [review]
Looks great!

**Comment by Nokel81** [comment]
Thanks for merging this, may I ask what if there is an eta for 7.29.5 (or more likely 7.30.0)?

**Comment by ljharb** [comment]
it'd be 7.30, yes. and there's no ETA.

**Pull request merged**